synaptic_reconstruction/tools/morphology_widget.py

```diff
@@ -0,0 +1,293 @@
+            table_data (pd.DataFrame): The table data containing coordinates, radii, and properties.
+            name (str): Name of the layer.
+
+        Returns:
+            Shapes layer: The created Napari Shapes layer.
+        """
+        coords = table_data[['x', 'y']].to_numpy() if 'z' not in table_data.columns else table_data[['x', 'y', 'z']].to_numpy()
+        radii = table_data['radii'].to_numpy()
+
+        if coords.shape[1] == 2:
+            # For 2D data, create circular outlines using trigonometric functions
+            outlines = [
+                np.column_stack((
+                    coords[i, 0] + radii[i] * np.cos(np.linspace(0, 2 * np.pi, 100)),
+                    coords[i, 1] + radii[i] * np.sin(np.linspace(0, 2 * np.pi, 100))
+                )) for i in range(len(coords))
+            ]
+        elif coords.shape[1] == 3:
+            # For 3D data, create spherical outlines using latitude and longitude
+            theta = np.linspace(0, 2 * np.pi, 50)  # Longitude
+            phi = np.linspace(0, np.pi, 25)       # Latitude
+            sphere_points = np.array([
+                [
+                    coords[i, 0] + radii[i] * np.sin(p) * np.cos(t),
+                    coords[i, 1] + radii[i] * np.sin(p) * np.sin(t),
+                    coords[i, 2] + radii[i] * np.cos(p)
+                ]
+                for i in range(len(coords))
+                for t in theta for p in phi
+            ])
+            outlines = [
+                sphere_points[i * len(theta) * len(phi):(i + 1) * len(theta) * len(phi)]
+                for i in range(len(coords))
+            ]
+        else:
+            raise ValueError("Coordinate dimensionality must be 2 or 3.")
+
+        # Add the shapes layer with properties
+        layer = self.viewer.add_shapes(
+            outlines,
+            name=name,
+            shape_type="polygon",  # Use "polygon" for closed shapes like circles
+            edge_width=2,
+            edge_color="red",
+            face_color="transparent",
+            blending="additive",
+            properties=table_data.to_dict(orient='list'),  # Attach table data as properties
+        )
+        return layer
+
+    def _to_table_data(self, coords, radii, props):
+        """
+        Create a table of data including coordinates, radii, and intensity statistics.
+
+        Args:
+            coords (np.ndarray): Array of 2D or 3D coordinates.
+            radii (np.ndarray): Array of radii corresponding to the coordinates.
+            props (list): List of properties containing intensity statistics.
+
+        Returns:
+            pd.DataFrame: The formatted table data.
+        """
+        assert len(coords) == len(radii), f"Shape mismatch: {coords.shape}, {radii.shape}"
+        
+        # Define columns based on dimension (2D or 3D)
+        col_names = ['x', 'y'] if coords.shape[1] == 2 else ['x', 'y', 'z']
+        table_data = {
+            'index': np.arange(len(coords)),
```

Comment:
We need to keep track of the `label_id` instead of the index. It can be accessed via the `prop.label` attribute (in the list returned by regionprops).